### ISSUE TITLE
Incorrect sql query after update to 4.4.4 or higher

### ISSUE THREAD

**Issue by ramirus** [mikro-orm/mikro-orm#1727]
**Describe the bug**
I try to count tasks with filter:
```
const mappedFilters = {
  user: { $or: [{ uid: user }, { name: { $like: `%${user}%` } }] }
}
em.count(Task, mappedFilters);
```
```
[app] error: select count(distinct `e0`.`id`) as `count` from `task` as `e0` left join `user` as `e1` on `e0`.`user_uid` = `e1`.`uid` left join `place` as `e2` on `e0`.`place_id` = `e2`.`id` where (`e0`.`uid` = '1234' or `e1`.`name` like '%1234%') - (conn=668, no: 1054, SQLState: 42S22) Unknown column 'e0.uid' in 'where clause'
```

**Expected behavior**
```
select count(distinct `e0`.`id`) as `count` from `task` as `e0` left join `user` as `e1` on `e0`.`user_uid` = `e1`.`uid` left join 
`place` as `e2` on `e0`.`place_id` = `e2`.`id` where (`e1`.`uid` = '1234' or `e1`.`name` like '%1234%')
```

**Versions**

| Dependency | Version |
| - | - |
| node | 15.14.0 |
| typescript | 4.1.3 |
| mikro-orm | 4.4.4 or higher |
| @mikro-orm/mariadb | 4.4.4 or higher  |

**Comment by B4nan** [comment]
Please provide something reproducible, e.g. we need full entity definitions, and ideally some data as well. Also have you tried with the latest version (v4.5.4, shipped ~2 hours ago)?

**Comment by ramirus** [comment]
Yes, also tried latest.
This is my entity definitions:
```@Entity()
export class Task {
  @Field(() => ID)
  @PrimaryKey()
  id: TaskId;

  @Field(() => User, { nullable: true })
  @ManyToOne({
    entity: () => User,
    nullable: true,
    cascade: [Cascade.PERSIST],
  })
  user?: User | null;

  @Field(() => Place, { nullable: true })
  @ManyToOne({
    entity: () => Place,
    nullable: true,
    onUpdateIntegrity: "cascade",
    onDelete: "restrict",
  })
  place?: Place | null;
  ...
  }

@ObjectType()
@Entity()
export class User {
  @Field(() => ID)
  @PrimaryKey({ length: 60 })
  uid: UserUID;

  @Field({ nullable: true })
  @Property({
    nullable: true,
  })
  name?: string;
...
}

@ObjectType()
@Entity()
export class Place {
  @Field(() => ID)
  @PrimaryKey({ length: 60 })
  id: PlaceId;

  @Field({ nullable: true })
  @Property({ nullable: true })
  name?: string;
...
}
```

**Comment by B4nan** [comment]
Workaround is to move the operator to top level:

```ts
    const query = {
      $or: [
        { user: { uid: 123 } },
        { user: { name: { $like: `%123%` } } },
      ],
    };
    await em.count(Task, query);
```

**Comment by B4nan** [comment]
Fixed via 38775e68b98c927e0bb5ae5ece38bb53f797c30c

**Comment by B4nan** [comment]
`4.5.5-dev.3`
